codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/protocol/aws/AwsJson10ProtocolGenerator.java

```diff
@@ -47,16 +48,29 @@ public ApplicationProtocol getApplicationProtocol() {
     @Override
     public void generateRequestSerializers(GenerationContext context) {
         var writer = context.getWriter().get();
-        var model = context.getModel();
-        var ops = model.getOperationShapes();
+        var ops = context.getModel().getOperationShapes();
+        for (var op : ops) {
+            var middle = new SerializeMiddleware(context, op, writer);
+            writer.write(middle.generate());
+            writer.write("\n");
+        }
+        generateSharedSerializers(context, writer, ops);
+    }
+
+    public void generateSharedSerializers(GenerationContext context, GoWriter writer, Set<OperationShape> ops) {
+        Set<Shape> shared = new java.util.HashSet<>(Set.of());
```

Comment:
You're creating an unnecessary set.

When you call  `Set.of()` you're creating an empty set. Then, when you call `new HashSet<>()`, you're creating another new set with an empty set as an argument. This is usually done when you want to copy all the elements of set A into set B 

```
Set<String> A = (some values) 
Set<String> B = new HashSet<>(A); // copy all the values already present in A to B
```

But you don't need that here, you just want an empty set. Just call `new HashSet<>()`.